apps/ledger-live-mobile/src/components/Nft/NftList.hook.ts

```diff
@@ -0,0 +1,108 @@
+import { decodeNftId } from "@ledgerhq/live-common/nft/nftId";
+import { useToasts } from "@ledgerhq/live-common/notifications/ToastProvider/index";
+import { ProtoNFT, NFTMetadata } from "@ledgerhq/types-live";
+import { useIsFocused, useNavigation } from "@react-navigation/native";
+import { v4 as uuid } from "uuid";
+import { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import {
+  hideNftCollection,
+  updateMainNavigatorVisibility,
+} from "../../actions/settings";
+import { track } from "../../analytics";
+import { NavigatorName, ScreenName } from "../../const";
+
+const TOAST_ID = "SUCCESS_HIDE";
+
+export function useNftList() {
+  const isFocused = useIsFocused();
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const { pushToast } = useToasts();
+  const navigation = useNavigation();
+
+  const [onMultiSelectMode, setMultiSelectMode] = useState<boolean>(false);
+
+  const [nftsToHide, setNftsToHide] = useState<ProtoNFT[]>([]);
+
+  // Multi Select ------------------------
+
+  const readOnlyModeAction = useCallback(() => {
+    setNftsToHide([]);
+    setMultiSelectMode(false);
+    dispatch(updateMainNavigatorVisibility(true));
+  }, [dispatch]);
+
+  // Detect change of screen
+  useEffect(() => {
+    if (!isFocused) {
+      readOnlyModeAction();
+    }
+  }, [readOnlyModeAction, isFocused]);
+
+  const onClickHide = useCallback(() => {
+    nftsToHide.forEach(nft => {
+      const { accountId } = decodeNftId(nft.id ?? "");
+      dispatch(hideNftCollection(`${accountId}|${nft.contract}`));
+    });
+    pushToast({
+      id: `${TOAST_ID}-${uuid()}`,
+      type: "success",
+      icon: "success",
+      title: t("wallet.nftGallery.filters.alertHide", {
+        count: nftsToHide.length,
+      }),
+    });
+
+    readOnlyModeAction();
+  }, [readOnlyModeAction, dispatch, nftsToHide, pushToast, t]);
+
+  const multiSelectModeAction = useCallback(() => {
+    setNftsToHide([]);
+    setMultiSelectMode(true);
+    dispatch(updateMainNavigatorVisibility(false));
```

Comment:
What happens if you quit the app right after this happens? (without finishing the selection)
Then restart it, won't the tab bar be hidden, putting the app in a blocked state? Because values of the `settings` part of the store are persisted. In my opinion even if you have some logic to "unhide it" automatically on mount (which could solve this issue), it's a bit dangerous to save that state persistently in the settings.

Reply:
Also for "safety" I'd also call a `dispatch(updateMainNavigatorVisibility(true)` in the cleanup function of a `useEffect` to ensure that we return to a normal state when this hook here is unmounted.

Reply:
https://user-images.githubusercontent.com/112866305/214809677-1bcde5bb-5544-40af-8c52-b0023d2423da.mov

Well yes you're right but seems working well

To solve that maybe create a new part of store that is not written persistently?

Reply:
You can use `appstate` for that I think

Reply:
Right now it works because the "crypto/nft tab" is persisted as well so the app always starts on that tab if you left it there, but we should not depend on this logic. The solution is to not persist that value.

On the other point I mentioned, where you should call `dispatch(updateMainNavigatorVisibility(true)` when the hook unmounts (so in the cleanup function), here's a (twisted) example with a video to illustrate:

1. user starts selecting
2. puts the app in background
3. opens a deeplink to Ledger (this could happen for many reasons
4. user is stuck on the page that the deeplink brought them to


https://user-images.githubusercontent.com/91890529/214820570-efa5a5bb-a228-4ff7-aff0-52d28518773b.mp4

Reply:
Another thing that would be a nice to have regarding this "selection mode" is to take into account "back button presses / gestures" (especially on Android) to cancel the selection mode.

There is an example of how it can be done in `TabBar/Transfer.tsx`. If you do it don't forget to return a boolean value in the listener function that you pass to the `BackHandler` (always return false is the selection mode is off, to let the event bubble up to other listeners and eventually to the native handler). Cf. https://reactnative.dev/docs/backhandler

Reply:
^ the reason we need this is that this is the expected behaviour when navigating, and since you're not doing a real navigation here (but a real one in terms of UX), back gesture behaviour is not implemented but would be expected by the user